tedana/decay.py

```diff
@@ -28,7 +28,153 @@ def mono_exp(tes, s0, t2star):
     return s0 * np.exp(-tes / t2star)
 
 
-def fit_decay(data, tes, mask, masksum, fittype):
+def fit_monoexponential(data_cat, echo_times, adaptive_mask):
+    """
+    Fit monoexponential decay model with nonlinear curve-fitting.
+
+    Parameters
+    ----------
+    data_cat
+    echo_times
+    adaptive_mask
+
+    Returns
+    -------
+    t2s_limited, s0_limited, t2s_full, s0_full
+    """
+    RepLGR.info("A monoexponential model was fit to the data at each voxel "
+                "using nonlinear model fitting in order to estimate T2* and S0 "
+                "maps. For each voxel, the value from the adaptive mask was "
+                "used to determine which echoes would be used to estimate T2* "
+                "and S0.")
```

Comment:
This is a bit tricky - the automated methods will have this, and then the paragraph from loglin fit, since it is called within this function. Maybe a flag in log lin to avoid appending the "loglin methods" if they were called from the fit_monoexponential function. Or perhaps even better, the it instead states something like "Log linear Estimates were used to initialize the non-linear fit. "

Reply:
Nice catch. I totally missed that.

Since the `fit_loglinear` function is now, in effect, pretty much private, I don't have a problem adding a flag. I'll also update the monoexponential report.